### DISCUSSION TITLE
Is there a way for another stimulus controller to change the selection?

### DISCUSSION THREAD

**Discussion by mmhan** [josefarias/hotwire_combobox#138] [Q&A]
I'd like to change the selection from another controller, is there a possibility for that?

**Comment by josefarias** [comment]
Interesting. What's the use case for this?

Lots of things need to happen when selecting an option, assuming the option is even available since the combobox would be closed and async options won't be loaded yet. I can't think of a straightforward, reliable way to do it from outside the combobox right now.

**Comment by mmhan** [comment]
The use case something like person name combobox field and type combobox field, I want to automatically select the type combobox field automatically based on the person's last record's type.

**Reply by josefarias** [reply]
@mmhan I think what you're looking for is the prefilled combobox functionality. Check out the prefilled combobox section of the docs at https://hotwirecombobox.com

Let me know if that doesn't work for you. Closing for now.

**Reply by mmhan** [reply]
Prefilled combobox can only be used if the type combobox isn't rendered yet. I'd like to be able to select a type from the type combobox after it has already been rendered.

**Reply by josefarias** [reply]
I see. Sorry to say there is no way to reliably select an option from outside the combobox in all scenarios right now.

I have no plans to implement this myself but if anyone wants to attempt a PR they're welcome to. Might be good to share an implementation plan first though.

**Reply by josefarias** [reply]
@mmhan maybe if you put your combobox inside a turbo frame and have that external action change some state on the server you could re-render the combobox with a different prefilled value when an action is taken.

**Reply by josefarias** [reply]
Could use a turbo stream instead, too
